block_my_feedback.php

```diff
@@ -153,10 +154,30 @@ public static function fetch_marking(stdClass $user): ?array {
                     $assess = new stdClass;
                     $assess->cmid = $cmid;
                     $assess->modname = $mod->modname;
-                    // Get due date and require marking.
-                    $assess = self::get_mod_data($mod, $assess);
+                    // Turnitin assessments may have multiple parts, so treat them separately.
```

Comment:
don't we have some code that does this kind of thing in feeedback tracker that could be moved into a common reusable function?

Reply:
As there are similarities the code for treating parts of Turnitin modules in feedback_tracker but there much more data is calculated, for example feedback due dates, that are not used in the block.
I think it is better to keep them separated.